### PR TITLE
Various fixes to `ReturnVoidInsteadOfEmptyTuple`.

### PR DESCRIPTION
**Pull request by allevato**
- Handle `()` in closure signature return types.
- Handle `()` when nested in the argument lists of function types and closures.
- Ensure leading/trailing trivia of the original `()` is preserved.
- Don't replace `()` if it has internal comments, like `(/*foo*/)`.

### CODE REVIEW

**Comment by allevato** [comment]
cc @dylansturg

File: Sources/SwiftFormatRules/ReturnVoidInsteadOfEmptyTuple.swift [hunk 1]
```diff
@@ -25,14 +25,82 @@ public final class ReturnVoidInsteadOfEmptyTuple: SyntaxFormatRule {
...
+    let voidKeyword = makeVoidIdentifierType(toReplace: returnType)
+    return TypeSyntax(node.withArguments(arguments).withReturnType(TypeSyntax(voidKeyword)))
+  }
+
+  public override func visit(_ node: ClosureSignatureSyntax) -> Syntax {
+    guard let output = node.output,
+      let returnType = output.returnType.as(TupleTypeSyntax.self),
+      returnType.elements.count == 0
+    else {
+      return Syntax(node)
```
[the remark this answers is not in the document]
Comment (allevato) [reply Sources/SwiftFormatRules/ReturnVoidInsteadOfEmptyTuple.swift]: Oops, thanks for catching that. I also didn't have a test case that would have caught it (a non-empty-tuple return type with an argument list that has a function type returning an empty tuple). Fixed!
